plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -688,20 +688,18 @@ public void finishInsertTable(ConnectorSession session, JdbcOutputTableHandle ha
                 .map(this::quoted)
                 .collect(joining(", "));
         String insertSql = format("INSERT INTO %s (%s) SELECT %s FROM %s", targetTable, columnNames, columnNames, temporaryTable);
-        String cleanupSql = "DROP TABLE " + temporaryTable;
 
         try (Connection connection = getConnection(session, handle)) {
             execute(connection, insertSql);
         }
         catch (SQLException e) {
             throw new TrinoException(JDBC_ERROR, e);
         }
-
-        try (Connection connection = getConnection(session, handle)) {
-            execute(connection, cleanupSql);
-        }
-        catch (SQLException e) {
-            log.warn(e, "Failed to cleanup temporary table: %s", temporaryTable);
+        finally {
+            JdbcTableHandle temporaryTableHandle = new JdbcTableHandle(
+                    new SchemaTableName(handle.getSchemaName(), handle.getTemporaryTableName()),
+                    new RemoteTableName(Optional.ofNullable(handle.getCatalogName()), Optional.ofNullable(handle.getSchemaName()), handle.getTemporaryTableName()));
```

Comment:
can handle.getTemporaryTableName() != remote name of the table (eg in case it gets upper-cased or lower-cased by the remote side)?

Reply:
@findepi, I think this @kokosing's comment addresses your's question.

Reply:
No, `JdbcOutputTableHandle` is populated`io.trino.plugin.jdbc.BaseJdbcClient#beginInsertTable`
```
return new JdbcOutputTableHandle(
                    catalog,
                    remoteSchema,
                    remoteTable,
                    columnNames.build(),
                    columnTypes.build(),
                    Optional.of(jdbcColumnTypes.build()),
                    remoteTemporaryTableName);
```